src/utils/string.ts

```diff
@@ -21,3 +21,7 @@ export const getSymbolSlug = (token: Currency | Token | undefined) =>
 export const getNetworkSlug = (chainId: ChainId | undefined) => {
   return chainId ? NETWORKS_INFO[chainId].route : ''
 }
+
+export const isInEnum = <T extends Record<string, string>>(str: string, enumParam: T): str is T[keyof T] => {
```

Comment:
we can update it to support both `enum number` and `enum string`?

Reply:
I cant, you can try do it by yourself: [link](https://www.typescriptlang.org/play?#code/PTAECUFMEcFcEsBOkC2kB2AXAXAKBKAIwB0oAxgPboDOFANpMXRQOajKayLrkWLJlMdAJ7tI1WHUz4wAJlKZhAB0iglyACbwyAQ0yr41UAHc+Aa3joWM0LdABeAHygAFhQBukRKEwVQAA3gAM1AACkMASXQAUXRYFFCdQgAaAGVMREsWWPiASlzQAG8kgF9QSDpqVWLCEv9kn2VVdUgtXX1QajdjHxc9UCCkakwi0tBDAMoaEaTsUHTMqxyUf1wp4fHqKOWHUAAeAFVygA99dA0jYcW2AB9QOJQAIy8GgBUTs4uISEpEDT2rlkGoCrKA7gdHI5Qlc5gcGhh4gAFHSIHQoOavXJzK6bUCvADaZkgwgoIVeAF0HM5CrhbMEwooVKTOhkHPZ7KAAOQglicgocLg8ADyjwAVj9MMR3Do6LBxKEEShkaiULliJYyLKNPKrrlaeMQqFGZBmTj2RzOQ9nog+WJONwWd5LOUHsq0fqBQ6gjKqrgSrhcIr5hksjsabYAIK7Tk6WOc5L6gBC0ceqfjfrWVA2s0dWWjsZ0nMz01Aj0I2JDoItZBrRfW9EYzBYoU5kRiD0SKWD12WuXjm22HaSDQWoYe+VABAycuLtAYTFYLbby1CZZHley4-7y47a+7Y7yBQI3sqkFw9PCW3b8U765744KNTKwAAVI0VGpNNo9KpR0sHqAL5gBUVSjLU56GjuN57n+m6HkUZZlCB1SIQGQYAHLxNaYb6lGHKEAADARCa2MmHKyERCb+usMyyHMVpeLsFEEbOIyPHR9xYYxHIAMxEbODYLs2rZXiuOiyA0mFPF4vbbqJQ4SaAUnYQ+k5gNOZ71vOTZLvJ0GKcpMlbg0UEJOxklcYgvZHmAJ6+he4njDwhlWeOoyyEhp7uf6F6mYkBmWdZ3mTm+xqfq034dC5OxAeUXnFB5aEAQAEpA+iIKwGAmrA1A4ZGuyEcRSb5nGVGsaAOg8fRlkFfxNGllVnHSd45F1VmIxkI1PIlYW5UaF1G4pmmAnaYuImDjelUNKl6WZeg2W5cZA7XgkU2gDNXhzQtQVTogM5aY2Y1+Y8PHTWlm0sFlFA5bJJl6WZp3redGWXfN12LfBu37Vmgk6eNK2hJ1Z2za921LX5QNPSDV03apX2aT9o3CX5-XAxdMMfaqckTQkqNQ+jb2w59tk+mevn3Ykj0bS9GNBcUPGeaB9M+ZBFMnWjNOE5jE6FCdjMoQzEFhBDVPPVt71051-NFFLQuXjjoR49T4tE6qj79dLhQa7gQA)